{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# 1. Convert the raw 23andMe data to vcf\n",
    "\n",
    "We will remove all SNPs corresponding to deletions and insertions, to make the file compatible with annotation tools\n",
    "\n",
    "`../plink_mac_20201019/plink --23file data/SNP_raw_v4_Full_20170514175358.txt --recode vcf --out snps_clean --output-chr MT --snps-only just-acgt`\n",
    "\n",
    "We interested only clinically relevant SNPs, so we will intersect result with vcf from ClinVar database (https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/clinvar.vcf.gz).\n",
    "\n",
    "Download and unzip:\n",
    "\n",
    "`wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/clinvar.vcf.gz`\n",
    "\n",
    "`gzip -d clinvar.vcf.gz`\n",
    "\n",
    "Intersect:\n",
    "\n",
    "`bedtools intersect -a snps_clean.vcf -b clinvar.vcf  > intersected.vcf`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Annotation\n",
    "\n",
    "We will use VEP (Variant Effect Predictor) online version (http://grch37.ensembl.org/Homo_sapiens/Tools/VEP).\n",
    "\n",
    "Input:\n",
    "\n",
    "<img src=\"img/vep.png\">\n",
    "\n",
    "Result in `data/vep_result.txt`. Now look at result:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000378536.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000478223.2</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000507179.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000508416.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2234167</td>\n",
       "      <td>1:2494330-2494330</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>ENSG00000157873</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000355716.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #Uploaded_variation           Location Allele  \\\n",
       "0           rs2843159  1:2235672-2235672      T   \n",
       "1           rs2843159  1:2235672-2235672      T   \n",
       "2           rs2843159  1:2235672-2235672      T   \n",
       "3           rs2843159  1:2235672-2235672      T   \n",
       "4           rs2234167  1:2494330-2494330      G   \n",
       "\n",
       "                                    Consequence    IMPACT    SYMBOL  \\\n",
       "0                                intron_variant  MODIFIER       SKI   \n",
       "1                       downstream_gene_variant  MODIFIER       SKI   \n",
       "2  intron_variant,non_coding_transcript_variant  MODIFIER       SKI   \n",
       "3                       downstream_gene_variant  MODIFIER       SKI   \n",
       "4                              missense_variant  MODERATE  TNFRSF14   \n",
       "\n",
       "              Gene Feature_type            Feature               BIOTYPE  ...  \\\n",
       "0  ENSG00000157933   Transcript  ENST00000378536.4        protein_coding  ...   \n",
       "1  ENSG00000157933   Transcript  ENST00000478223.2  processed_transcript  ...   \n",
       "2  ENSG00000157933   Transcript  ENST00000507179.1       retained_intron  ...   \n",
       "3  ENSG00000157933   Transcript  ENST00000508416.1  processed_transcript  ...   \n",
       "4  ENSG00000157873   Transcript  ENST00000355716.4        protein_coding  ...   \n",
       "\n",
       "       AF CLIN_SIG SOMATIC PHENO             PUBMED MOTIF_NAME MOTIF_POS  \\\n",
       "0  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "1  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "2  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "3  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "4       -        -       -     -                  -          -         -   \n",
       "\n",
       "  HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "0            -                  -                     -  \n",
       "1            -                  -                     -  \n",
       "2            -                  -                     -  \n",
       "3            -                  -                     -  \n",
       "4            -                  -                     -  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('data/vep_result.txt', sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will look at CLIN_SIG and try to find pathogenic and risk_factor SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'-',\n",
       " 'affects',\n",
       " 'association',\n",
       " 'association,benign',\n",
       " 'association_not_found',\n",
       " 'benign',\n",
       " 'benign,_other,benign',\n",
       " 'benign,affects',\n",
       " 'benign,benign/likely_benign',\n",
       " 'benign,benign/likely_benign,pathogenic',\n",
       " 'benign,drug_response',\n",
       " 'benign,likely_benign',\n",
       " 'benign,likely_benign,other',\n",
       " 'benign,likely_pathogenic',\n",
       " 'benign,not_provided',\n",
       " 'benign,not_provided,drug_response',\n",
       " 'benign,not_provided,likely_benign',\n",
       " 'benign,pathogenic',\n",
       " 'benign,risk_factor',\n",
       " 'benign/likely_benign',\n",
       " 'benign/likely_benign,benign',\n",
       " 'benign/likely_benign,benign,likely_benign',\n",
       " 'conflicting_interpretations_of_pathogenicity,benign',\n",
       " 'drug_response',\n",
       " 'drug_response,benign',\n",
       " 'drug_response,benign,_other',\n",
       " 'drug_response,benign,risk_factor',\n",
       " 'drug_response,benign/likely_benign,benign,likely_benign',\n",
       " 'drug_response,likely_benign,benign',\n",
       " 'drug_response,not_provided',\n",
       " 'likely_benign',\n",
       " 'likely_benign,benign',\n",
       " 'likely_benign,benign,risk_factor',\n",
       " 'likely_benign,benign/likely_benign,pathogenic,benign',\n",
       " 'likely_benign,pathogenic',\n",
       " 'likely_pathogenic,benign',\n",
       " 'not_provided',\n",
       " 'not_provided,benign',\n",
       " 'not_provided,pathogenic,likely_benign,benign',\n",
       " 'other',\n",
       " 'other,benign',\n",
       " 'pathogenic',\n",
       " 'pathogenic,_risk_factor',\n",
       " 'pathogenic,benign',\n",
       " 'pathogenic,benign/likely_benign,benign',\n",
       " 'pathogenic,likely_benign,benign',\n",
       " 'pathogenic,risk_factor',\n",
       " 'protective',\n",
       " 'protective,likely_benign,benign',\n",
       " 'protective,pathogenic',\n",
       " 'risk_factor',\n",
       " 'risk_factor,association,benign',\n",
       " 'risk_factor,benign',\n",
       " 'risk_factor,benign,drug_response',\n",
       " 'risk_factor,benign,likely_benign',\n",
       " 'risk_factor,drug_response',\n",
       " 'risk_factor,likely_benign',\n",
       " 'risk_factor,pathogenic',\n",
       " 'uncertain_significance',\n",
       " 'uncertain_significance,benign',\n",
       " 'uncertain_significance,benign,drug_response,risk_factor',\n",
       " 'uncertain_significance,likely_benign,conflicting_interpretations_of_pathogenicity'}"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clin_sigs = set()\n",
    "\n",
    "for c in df['CLIN_SIG'].unique():\n",
    "    if type(c) == str:\n",
    "        c_a = c.split('|')\n",
    "        clin_sigs.update(c_a)\n",
    "\n",
    "clin_sigs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will look at pathogenic tag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "#Uploaded_variation\n",
       "i3000469                                            risk_factor\n",
       "i5005436                                             pathogenic\n",
       "i5006568      likely_benign,benign/likely_benign,pathogenic,...\n",
       "i6007787                                     risk_factor,benign\n",
       "i6015290      uncertain_significance,likely_benign,conflicti...\n",
       "i6015729                 pathogenic,benign/likely_benign,benign\n",
       "i6058143                       drug_response,benign,risk_factor\n",
       "i6058764                                  protective,pathogenic\n",
       "i6059141                       likely_benign,benign,risk_factor\n",
       "i6060296      conflicting_interpretations_of_pathogenicity,b...\n",
       "rs10151259         not_provided,pathogenic,likely_benign,benign\n",
       "rs1024611                                pathogenic,risk_factor\n",
       "rs1042503                              likely_pathogenic,benign\n",
       "rs1049296                        risk_factor,association,benign\n",
       "rs11085825                             benign,likely_pathogenic\n",
       "rs11558492                      pathogenic,likely_benign,benign\n",
       "rs1169288                                    benign,risk_factor\n",
       "rs12150220                                          risk_factor\n",
       "rs13266634                                          risk_factor\n",
       "rs16879498                                    benign,pathogenic\n",
       "rs1800435                              likely_benign,pathogenic\n",
       "rs1801197                                           risk_factor\n",
       "rs1801274                      drug_response,benign,risk_factor\n",
       "rs1801275                                           risk_factor\n",
       "rs1801394     uncertain_significance,benign,drug_response,ri...\n",
       "rs1801968                      likely_benign,benign,risk_factor\n",
       "rs2004640                                risk_factor,pathogenic\n",
       "rs2073658                                           risk_factor\n",
       "rs2184026                                           risk_factor\n",
       "rs2239704                                           risk_factor\n",
       "rs2241880                                    risk_factor,benign\n",
       "rs2281845                                    risk_factor,benign\n",
       "rs2301612                                     pathogenic,benign\n",
       "rs2306283     conflicting_interpretations_of_pathogenicity,b...\n",
       "rs231775                                     benign,risk_factor\n",
       "rs3755319                                            pathogenic\n",
       "rs4402960                                           risk_factor\n",
       "rs460897                                pathogenic,_risk_factor\n",
       "rs4673                                 likely_pathogenic,benign\n",
       "rs4880                         risk_factor,benign,drug_response\n",
       "rs4961                                risk_factor,drug_response\n",
       "rs5174                                              risk_factor\n",
       "rs5186                                       risk_factor,benign\n",
       "rs61747071                     risk_factor,benign,likely_benign\n",
       "rs6265                                       risk_factor,benign\n",
       "rs6280                                risk_factor,likely_benign\n",
       "rs6339                   benign,benign/likely_benign,pathogenic\n",
       "rs6504649                                           risk_factor\n",
       "rs699                                        benign,risk_factor\n",
       "rs763110                                            risk_factor\n",
       "rs7794745                                           risk_factor\n",
       "rs909253                                            risk_factor\n",
       "Name: CLIN_SIG, dtype: object"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_for = ['pathogenic', 'risk_factor']\n",
    "pathogenic_df = df[df['CLIN_SIG'].str.contains('|'.join(search_for))].groupby('#Uploaded_variation').first()\n",
    "pathogenic_df = pathogenic_df[['Location', 'Allele', 'Codons', 'CLIN_SIG']]\n",
    "pathogenic_df.CLIN_SIG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Codons</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>i3000469</th>\n",
       "      <td>2:138759649-138759649</td>\n",
       "      <td>T</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i5005436</th>\n",
       "      <td>6:32008312-32008312</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i5006568</th>\n",
       "      <td>2:71829924-71829924</td>\n",
       "      <td>G</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>likely_benign,benign/likely_benign,pathogenic,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6007787</th>\n",
       "      <td>2:234183368-234183368</td>\n",
       "      <td>G</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6015290</th>\n",
       "      <td>14:23887607-23887607</td>\n",
       "      <td>T</td>\n",
       "      <td>aaC/aaA</td>\n",
       "      <td>uncertain_significance,likely_benign,conflicti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6015729</th>\n",
       "      <td>1:156848918-156848918</td>\n",
       "      <td>T</td>\n",
       "      <td>Cat/Tat</td>\n",
       "      <td>pathogenic,benign/likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6058143</th>\n",
       "      <td>1:161479745-161479745</td>\n",
       "      <td>G</td>\n",
       "      <td>cAt/cGt</td>\n",
       "      <td>drug_response,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6058764</th>\n",
       "      <td>16:27356203-27356203</td>\n",
       "      <td>G</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>protective,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6059141</th>\n",
       "      <td>8:133909974-133909974</td>\n",
       "      <td>G</td>\n",
       "      <td>Atg/Gtg</td>\n",
       "      <td>likely_benign,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6060296</th>\n",
       "      <td>19:13010520-13010520</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity,b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs10151259</th>\n",
       "      <td>14:21790040-21790040</td>\n",
       "      <td>T</td>\n",
       "      <td>Gct/Tct</td>\n",
       "      <td>not_provided,pathogenic,likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1024611</th>\n",
       "      <td>17:32579788-32579788</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1042503</th>\n",
       "      <td>12:103246700-103246700</td>\n",
       "      <td>T</td>\n",
       "      <td>gtG/gtA</td>\n",
       "      <td>likely_pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1049296</th>\n",
       "      <td>3:133494354-133494354</td>\n",
       "      <td>T</td>\n",
       "      <td>Cct/Tct</td>\n",
       "      <td>risk_factor,association,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs11085825</th>\n",
       "      <td>19:13007458-13007458</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs11558492</th>\n",
       "      <td>1:231408091-231408091</td>\n",
       "      <td>G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>pathogenic,likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1169288</th>\n",
       "      <td>12:121416650-121416650</td>\n",
       "      <td>C</td>\n",
       "      <td>Atc/Ctc</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs12150220</th>\n",
       "      <td>17:5485367-5485367</td>\n",
       "      <td>T</td>\n",
       "      <td>cTc/cAc</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs13266634</th>\n",
       "      <td>8:118184783-118184783</td>\n",
       "      <td>T</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs16879498</th>\n",
       "      <td>6:49580247-49580247</td>\n",
       "      <td>T</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>benign,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1800435</th>\n",
       "      <td>9:116153891-116153891</td>\n",
       "      <td>G</td>\n",
       "      <td>aaG/aaC</td>\n",
       "      <td>likely_benign,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801197</th>\n",
       "      <td>7:93055753-93055753</td>\n",
       "      <td>G</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801274</th>\n",
       "      <td>1:161479745-161479745</td>\n",
       "      <td>G</td>\n",
       "      <td>cAt/cGt</td>\n",
       "      <td>drug_response,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801275</th>\n",
       "      <td>16:27374400-27374400</td>\n",
       "      <td>G</td>\n",
       "      <td>cAg/cGg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801394</th>\n",
       "      <td>5:7870973-7870973</td>\n",
       "      <td>G</td>\n",
       "      <td>atA/atG</td>\n",
       "      <td>uncertain_significance,benign,drug_response,ri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801968</th>\n",
       "      <td>9:132580901-132580901</td>\n",
       "      <td>G</td>\n",
       "      <td>Gat/Cat</td>\n",
       "      <td>likely_benign,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2004640</th>\n",
       "      <td>7:128578301-128578301</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2073658</th>\n",
       "      <td>1:161010762-161010762</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2184026</th>\n",
       "      <td>9:101304348-101304348</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2239704</th>\n",
       "      <td>6:31540141-31540141</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2241880</th>\n",
       "      <td>2:234183368-234183368</td>\n",
       "      <td>G</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2281845</th>\n",
       "      <td>1:201081943-201081943</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2301612</th>\n",
       "      <td>9:136301982-136301982</td>\n",
       "      <td>G</td>\n",
       "      <td>Caa/Gaa</td>\n",
       "      <td>pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2306283</th>\n",
       "      <td>12:21329738-21329738</td>\n",
       "      <td>G</td>\n",
       "      <td>Aat/Gat</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity,b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs231775</th>\n",
       "      <td>2:204732714-204732714</td>\n",
       "      <td>G</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs3755319</th>\n",
       "      <td>2:234667582-234667582</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4402960</th>\n",
       "      <td>3:185511687-185511687</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs460897</th>\n",
       "      <td>1:196716319-196716319</td>\n",
       "      <td>T</td>\n",
       "      <td>tCg/tTg</td>\n",
       "      <td>pathogenic,_risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4673</th>\n",
       "      <td>16:88713236-88713236</td>\n",
       "      <td>G</td>\n",
       "      <td>Tac/Cac</td>\n",
       "      <td>likely_pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4880</th>\n",
       "      <td>6:160113872-160113872</td>\n",
       "      <td>G</td>\n",
       "      <td>gTt/gCt</td>\n",
       "      <td>risk_factor,benign,drug_response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4961</th>\n",
       "      <td>4:2906707-2906707</td>\n",
       "      <td>T</td>\n",
       "      <td>Ggg/Tgg</td>\n",
       "      <td>risk_factor,drug_response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs5174</th>\n",
       "      <td>1:53712727-53712727</td>\n",
       "      <td>T</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs5186</th>\n",
       "      <td>3:148459988-148459988</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs61747071</th>\n",
       "      <td>16:53720436-53720436</td>\n",
       "      <td>T</td>\n",
       "      <td>Gct/Act</td>\n",
       "      <td>risk_factor,benign,likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6265</th>\n",
       "      <td>11:27679916-27679916</td>\n",
       "      <td>T</td>\n",
       "      <td>Gtg/Atg</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6280</th>\n",
       "      <td>3:113890815-113890815</td>\n",
       "      <td>T</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>risk_factor,likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6339</th>\n",
       "      <td>1:156848946-156848946</td>\n",
       "      <td>T</td>\n",
       "      <td>gGt/gTt</td>\n",
       "      <td>benign,benign/likely_benign,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6504649</th>\n",
       "      <td>17:48437456-48437456</td>\n",
       "      <td>G</td>\n",
       "      <td>aCa/aGa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs699</th>\n",
       "      <td>1:230845794-230845794</td>\n",
       "      <td>G</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs763110</th>\n",
       "      <td>1:172627498-172627498</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs7794745</th>\n",
       "      <td>7:146489606-146489606</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs909253</th>\n",
       "      <td>6:31540313-31540313</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Location Allele   Codons  \\\n",
       "#Uploaded_variation                                           \n",
       "i3000469              2:138759649-138759649      T  aCa/aTa   \n",
       "i5005436                6:32008312-32008312      T        -   \n",
       "i5006568                2:71829924-71829924      G  Atc/Gtc   \n",
       "i6007787              2:234183368-234183368      G  Act/Gct   \n",
       "i6015290               14:23887607-23887607      T  aaC/aaA   \n",
       "i6015729              1:156848918-156848918      T  Cat/Tat   \n",
       "i6058143              1:161479745-161479745      G  cAt/cGt   \n",
       "i6058764               16:27356203-27356203      G  Atc/Gtc   \n",
       "i6059141              8:133909974-133909974      G  Atg/Gtg   \n",
       "i6060296               19:13010520-13010520      G        -   \n",
       "rs10151259             14:21790040-21790040      T  Gct/Tct   \n",
       "rs1024611              17:32579788-32579788      G        -   \n",
       "rs1042503            12:103246700-103246700      T  gtG/gtA   \n",
       "rs1049296             3:133494354-133494354      T  Cct/Tct   \n",
       "rs11085825             19:13007458-13007458      T        -   \n",
       "rs11558492            1:231408091-231408091      G  gAt/gGt   \n",
       "rs1169288            12:121416650-121416650      C  Atc/Ctc   \n",
       "rs12150220               17:5485367-5485367      T  cTc/cAc   \n",
       "rs13266634            8:118184783-118184783      T  Cgg/Tgg   \n",
       "rs16879498              6:49580247-49580247      T  Gtt/Att   \n",
       "rs1800435             9:116153891-116153891      G  aaG/aaC   \n",
       "rs1801197               7:93055753-93055753      G  cTg/cCg   \n",
       "rs1801274             1:161479745-161479745      G  cAt/cGt   \n",
       "rs1801275              16:27374400-27374400      G  cAg/cGg   \n",
       "rs1801394                 5:7870973-7870973      G  atA/atG   \n",
       "rs1801968             9:132580901-132580901      G  Gat/Cat   \n",
       "rs2004640             7:128578301-128578301      T        -   \n",
       "rs2073658             1:161010762-161010762      T        -   \n",
       "rs2184026             9:101304348-101304348      T        -   \n",
       "rs2239704               6:31540141-31540141      C        -   \n",
       "rs2241880             2:234183368-234183368      G  Act/Gct   \n",
       "rs2281845             1:201081943-201081943      T        -   \n",
       "rs2301612             9:136301982-136301982      G  Caa/Gaa   \n",
       "rs2306283              12:21329738-21329738      G  Aat/Gat   \n",
       "rs231775              2:204732714-204732714      G  Acc/Gcc   \n",
       "rs3755319             2:234667582-234667582      C        -   \n",
       "rs4402960             3:185511687-185511687      T        -   \n",
       "rs460897              1:196716319-196716319      T  tCg/tTg   \n",
       "rs4673                 16:88713236-88713236      G  Tac/Cac   \n",
       "rs4880                6:160113872-160113872      G  gTt/gCt   \n",
       "rs4961                    4:2906707-2906707      T  Ggg/Tgg   \n",
       "rs5174                  1:53712727-53712727      T  cGa/cAa   \n",
       "rs5186                3:148459988-148459988      C        -   \n",
       "rs61747071             16:53720436-53720436      T  Gct/Act   \n",
       "rs6265                 11:27679916-27679916      T  Gtg/Atg   \n",
       "rs6280                3:113890815-113890815      T  Ggc/Agc   \n",
       "rs6339                1:156848946-156848946      T  gGt/gTt   \n",
       "rs6504649              17:48437456-48437456      G  aCa/aGa   \n",
       "rs699                 1:230845794-230845794      G  aTg/aCg   \n",
       "rs763110              1:172627498-172627498      T        -   \n",
       "rs7794745             7:146489606-146489606      T        -   \n",
       "rs909253                6:31540313-31540313      G        -   \n",
       "\n",
       "                                                              CLIN_SIG  \n",
       "#Uploaded_variation                                                     \n",
       "i3000469                                                   risk_factor  \n",
       "i5005436                                                    pathogenic  \n",
       "i5006568             likely_benign,benign/likely_benign,pathogenic,...  \n",
       "i6007787                                            risk_factor,benign  \n",
       "i6015290             uncertain_significance,likely_benign,conflicti...  \n",
       "i6015729                        pathogenic,benign/likely_benign,benign  \n",
       "i6058143                              drug_response,benign,risk_factor  \n",
       "i6058764                                         protective,pathogenic  \n",
       "i6059141                              likely_benign,benign,risk_factor  \n",
       "i6060296             conflicting_interpretations_of_pathogenicity,b...  \n",
       "rs10151259                not_provided,pathogenic,likely_benign,benign  \n",
       "rs1024611                                       pathogenic,risk_factor  \n",
       "rs1042503                                     likely_pathogenic,benign  \n",
       "rs1049296                               risk_factor,association,benign  \n",
       "rs11085825                                    benign,likely_pathogenic  \n",
       "rs11558492                             pathogenic,likely_benign,benign  \n",
       "rs1169288                                           benign,risk_factor  \n",
       "rs12150220                                                 risk_factor  \n",
       "rs13266634                                                 risk_factor  \n",
       "rs16879498                                           benign,pathogenic  \n",
       "rs1800435                                     likely_benign,pathogenic  \n",
       "rs1801197                                                  risk_factor  \n",
       "rs1801274                             drug_response,benign,risk_factor  \n",
       "rs1801275                                                  risk_factor  \n",
       "rs1801394            uncertain_significance,benign,drug_response,ri...  \n",
       "rs1801968                             likely_benign,benign,risk_factor  \n",
       "rs2004640                                       risk_factor,pathogenic  \n",
       "rs2073658                                                  risk_factor  \n",
       "rs2184026                                                  risk_factor  \n",
       "rs2239704                                                  risk_factor  \n",
       "rs2241880                                           risk_factor,benign  \n",
       "rs2281845                                           risk_factor,benign  \n",
       "rs2301612                                            pathogenic,benign  \n",
       "rs2306283            conflicting_interpretations_of_pathogenicity,b...  \n",
       "rs231775                                            benign,risk_factor  \n",
       "rs3755319                                                   pathogenic  \n",
       "rs4402960                                                  risk_factor  \n",
       "rs460897                                       pathogenic,_risk_factor  \n",
       "rs4673                                        likely_pathogenic,benign  \n",
       "rs4880                                risk_factor,benign,drug_response  \n",
       "rs4961                                       risk_factor,drug_response  \n",
       "rs5174                                                     risk_factor  \n",
       "rs5186                                              risk_factor,benign  \n",
       "rs61747071                            risk_factor,benign,likely_benign  \n",
       "rs6265                                              risk_factor,benign  \n",
       "rs6280                                       risk_factor,likely_benign  \n",
       "rs6339                          benign,benign/likely_benign,pathogenic  \n",
       "rs6504649                                                  risk_factor  \n",
       "rs699                                               benign,risk_factor  \n",
       "rs763110                                                   risk_factor  \n",
       "rs7794745                                                  risk_factor  \n",
       "rs909253                                                   risk_factor  "
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will find this SNPs in dbSNP https://www.ncbi.nlm.nih.gov/snp/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 3. mtDNA haplogroup.\n",
    "\n",
    "To define mtDNA haplogroup we will use: https://dna.jameslick.com/mthap/\n",
    "\n",
    "Results:\n",
    "\n",
    "mthap version 0.19b (2015-05-11); haplogroup data version PhyloTree Build 17 (2016-02-18) +mods \n",
    "raw data source SNP_raw_v4_Full_20170514175358.txt (14MB)\n",
    "\n",
    "Found 3270 markers at 3268 positions covering 19.7% of mtDNA.\n",
    "\n",
    "\n",
    "Markers found (shown as differences to rCRS):\n",
    "\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 8860G\n",
    "HVR1:\n",
    "\n",
    "IMPORTANT NOTE: The above marker list is almost certainly incomplete due to limitations of genotyping technology and is not comparable to mtDNA sequencing results. It should not be used with services or tools that expect sequencing results, such as mitosearch.\n",
    "\n",
    "\n",
    "Best mtDNA Haplogroup Matches:\n",
    "\n",
    "1) H(T152C)\n",
    "\n",
    "Defining Markers for haplogroup H(T152C):\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H(T152C):\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C ⇨ H(T152C)\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "2) H1(T152C)\n",
    "\n",
    "Defining Markers for haplogroup H1(T152C):\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 3010A 4769G 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H1(T152C):\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 3010A ⇨ H1 ⇨ 152C ⇨ H1(T152C)\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "No-Calls(1): 3010A\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H\n",
    "\n",
    "Defining Markers for haplogroup H:\n",
    "HVR2: 263G\n",
    "CR: 750G 1438G 4769G 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H (plus extra markers):\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(5): 263G 750G 1438G 4769G 8860G\n",
    "Extras(1): 152C\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H69\n",
    "\n",
    "Defining Markers for haplogroup H69:\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4646C 4769G 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H69:\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C ⇨ H(T152C) ⇨ 4646C ⇨ H69\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 4646T\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H3(T152C)\n",
    "\n",
    "Defining Markers for haplogroup H3(T152C):\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 6776C 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H3(T152C):\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 6776C ⇨ H3 ⇨ 152C ⇨ H3(T152C)\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 6776T\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H16(T152C)\n",
    "\n",
    "Defining Markers for haplogroup H16(T152C):\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 8860G 10394T 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H16(T152C):\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 10394T ⇨ H16 ⇨ 152C ⇨ H16(T152C)\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 10394C\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H9\n",
    "\n",
    "Defining Markers for haplogroup H9:\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 8860G 13020C 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H9:\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C ⇨ H(T152C) ⇨ 13020C ⇨ H9\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 13020T\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H46\n",
    "\n",
    "Defining Markers for haplogroup H46:\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 2772T 4769G 8860G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H46:\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C ⇨ H(T152C) ⇨ 2772T ⇨ H46\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 2772C\n",
    "Untested(1): 15326\n",
    "\n",
    "\n",
    "3) H52\n",
    "\n",
    "Defining Markers for haplogroup H52:\n",
    "HVR2: 152C 263G\n",
    "CR: 750G 1438G 4769G 8860G 14220G 15326G\n",
    "HVR1:\n",
    "\n",
    "Marker path from rCRS to haplogroup H52:\n",
    "H2a2a1(rCRS) ⇨ 263G ⇨ H2a2a ⇨ 8860G 15326G ⇨ H2a2 ⇨ 750G ⇨ H2a ⇨ 4769G ⇨ H2 ⇨ 1438G ⇨ H ⇨ 152C ⇨ H(T152C) ⇨ 14220G ⇨ H52\n",
    "\n",
    "Imperfect Match. Your results contained differences with this haplogroup:\n",
    "Matches(6): 152C 263G 750G 1438G 4769G 8860G\n",
    "Mismatches(1): 14220A\n",
    "Untested(1): 15326"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 4.  Y-chromosome haplogroup\n",
    "\n",
    "To define Y-chromosome haplogroup we will use: https://ytree.morleydna.com/extractFromAutosomal\n",
    "\n",
    "Result:\n",
    "\n",
    "<img src=\"img/y.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 5. Nationality.\n",
    "\n",
    "We will use https://github.com/stevenliuyi/admix to define nationality"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting admix\n",
      "  Downloading admix-1.2.tar.gz (37.6 MB)\n",
      "\u001b[K     |████████████████████████████████| 37.6 MB 233 kB/s eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: numpy in ./venv/lib/python3.8/site-packages (from admix) (1.20.1)\n",
      "Collecting scipy\n",
      "  Using cached scipy-1.6.1-cp38-cp38-macosx_10_9_x86_64.whl (30.8 MB)\n",
      "Using legacy 'setup.py install' for admix, since package 'wheel' is not installed.\n",
      "Installing collected packages: scipy, admix\n",
      "    Running setup.py install for admix ... \u001b[?25ldone\n",
      "\u001b[?25hSuccessfully installed admix-1.2 scipy-1.6.1\n"
     ]
    }
   ],
   "source": [
    "!pip install admix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Admixture calculation models: K7b,K12b,globe13,world9,E11\n",
      "\n",
      "Calcuation is started...\n",
      "\n",
      "K7b\n",
      "South Asian: 2.47%\n",
      "West Asian: 25.71%\n",
      "Siberian: 2.00%\n",
      "African: 0.12%\n",
      "Southern: 21.19%\n",
      "Atlantic Baltic: 48.50%\n",
      "East Asian: 0.00%\n",
      "\n",
      "\n",
      "K12b\n",
      "Gedrosia: 8.22%\n",
      "Siberian: 1.57%\n",
      "Northwest African: 1.83%\n",
      "Southeast Asian: 0.00%\n",
      "Atlantic Med: 18.38%\n",
      "North European: 34.10%\n",
      "South Asian: 1.73%\n",
      "East African: 0.00%\n",
      "Southwest Asian: 8.18%\n",
      "East Asian: 0.00%\n",
      "Caucasus: 25.99%\n",
      "Sub Saharan: 0.00%\n",
      "\n",
      "\n",
      "globe13\n",
      "Siberian: 1.43%\n",
      "Amerindian: 0.25%\n",
      "West African: 0.00%\n",
      "Palaeo African: 0.27%\n",
      "Southwest Asian: 12.85%\n",
      "East Asian: 0.00%\n",
      "Mediterranean: 24.47%\n",
      "Australasian: 0.91%\n",
      "Artic: 0.54%\n",
      "West Asian: 20.26%\n",
      "North European: 37.56%\n",
      "South Asian: 1.45%\n",
      "East African: 0.02%\n",
      "\n",
      "\n",
      "world9\n",
      "Amerindian: 0.22%\n",
      "East Asian: 0.00%\n",
      "African: 0.27%\n",
      "Atlantic Baltic: 48.02%\n",
      "Australasian: 0.70%\n",
      "Siberian: 2.02%\n",
      "Caucasus Gedrosia: 26.30%\n",
      "Southern: 20.76%\n",
      "South Asian: 1.71%\n",
      "\n",
      "\n",
      "E11\n",
      "African: 2.02%\n",
      "European: 68.54%\n",
      "India: 22.17%\n",
      "Malay: 0.30%\n",
      "South Chinese Dai: 0.00%\n",
      "Southwest Chinese Yi: 0.00%\n",
      "East Chinese: 0.00%\n",
      "Japanese: 0.00%\n",
      "North Chinese Oroqen: 1.04%\n",
      "Yakut: 1.35%\n",
      "American: 4.57%\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!admix -f data/SNP_raw_v4_Full_20170514175358.txt -v 23andme -m K7b K12b globe13 world9 E11"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
