{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# 1. Convert the raw 23andMe data to vcf\n",
    "\n",
    "We will remove all SNPs corresponding to deletions and insertions, to make the file compatible with annotation tools\n",
    "\n",
    "`../plink_mac_20201019/plink --23file data/SNP_raw_v4_Full_20170514175358.txt --recode vcf --out snps_clean --output-chr MT --snps-only just-acgt`\n",
    "\n",
    "We interested only clinically relevant SNPs, so we will intersect result with vcf from ClinVar database (https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/clinvar.vcf.gz).\n",
    "\n",
    "Download and unzip:\n",
    "\n",
    "`wget https://ftp.ncbi.nlm.nih.gov/pub/clinvar/vcf_GRCh37/clinvar.vcf.gz`\n",
    "\n",
    "`gzip -d clinvar.vcf.gz`\n",
    "\n",
    "Intersect:\n",
    "\n",
    "`bedtools intersect -a snps_clean.vcf -b clinvar.vcf  > intersected.vcf`"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 2. Annotation\n",
    "\n",
    "We will use VEP (Variant Effect Predictor) online version (http://grch37.ensembl.org/Homo_sapiens/Tools/VEP).\n",
    "\n",
    "Input:\n",
    "\n",
    "<img src=\"img/vep.png\">\n",
    "\n",
    "Result in `data/vep_result.txt`. Now look at result:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000378536.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000478223.2</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000507179.1</td>\n",
       "      <td>retained_intron</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs2843159</td>\n",
       "      <td>1:2235672-2235672</td>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SKI</td>\n",
       "      <td>ENSG00000157933</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000508416.1</td>\n",
       "      <td>processed_transcript</td>\n",
       "      <td>...</td>\n",
       "      <td>0.3498</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>16327884,22984993</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs2234167</td>\n",
       "      <td>1:2494330-2494330</td>\n",
       "      <td>G</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TNFRSF14</td>\n",
       "      <td>ENSG00000157873</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000355716.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  #Uploaded_variation           Location Allele  \\\n",
       "0           rs2843159  1:2235672-2235672      T   \n",
       "1           rs2843159  1:2235672-2235672      T   \n",
       "2           rs2843159  1:2235672-2235672      T   \n",
       "3           rs2843159  1:2235672-2235672      T   \n",
       "4           rs2234167  1:2494330-2494330      G   \n",
       "\n",
       "                                    Consequence    IMPACT    SYMBOL  \\\n",
       "0                                intron_variant  MODIFIER       SKI   \n",
       "1                       downstream_gene_variant  MODIFIER       SKI   \n",
       "2  intron_variant,non_coding_transcript_variant  MODIFIER       SKI   \n",
       "3                       downstream_gene_variant  MODIFIER       SKI   \n",
       "4                              missense_variant  MODERATE  TNFRSF14   \n",
       "\n",
       "              Gene Feature_type            Feature               BIOTYPE  ...  \\\n",
       "0  ENSG00000157933   Transcript  ENST00000378536.4        protein_coding  ...   \n",
       "1  ENSG00000157933   Transcript  ENST00000478223.2  processed_transcript  ...   \n",
       "2  ENSG00000157933   Transcript  ENST00000507179.1       retained_intron  ...   \n",
       "3  ENSG00000157933   Transcript  ENST00000508416.1  processed_transcript  ...   \n",
       "4  ENSG00000157873   Transcript  ENST00000355716.4        protein_coding  ...   \n",
       "\n",
       "       AF CLIN_SIG SOMATIC PHENO             PUBMED MOTIF_NAME MOTIF_POS  \\\n",
       "0  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "1  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "2  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "3  0.3498        -       -     -  16327884,22984993          -         -   \n",
       "4       -        -       -     -                  -          -         -   \n",
       "\n",
       "  HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "0            -                  -                     -  \n",
       "1            -                  -                     -  \n",
       "2            -                  -                     -  \n",
       "3            -                  -                     -  \n",
       "4            -                  -                     -  \n",
       "\n",
       "[5 rows x 41 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv('data/vep_result.txt', sep='\\t')\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we will look at CLIN_SIG and try to find pathogenic and risk_factor SNPs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'-',\n",
       " '_other',\n",
       " '_risk_factor',\n",
       " 'affects',\n",
       " 'association',\n",
       " 'association_not_found',\n",
       " 'benign',\n",
       " 'benign/likely_benign',\n",
       " 'conflicting_interpretations_of_pathogenicity',\n",
       " 'drug_response',\n",
       " 'likely_benign',\n",
       " 'likely_pathogenic',\n",
       " 'not_provided',\n",
       " 'other',\n",
       " 'pathogenic',\n",
       " 'protective',\n",
       " 'risk_factor',\n",
       " 'uncertain_significance'}"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clin_sigs = set()\n",
    "\n",
    "for c in df.CLIN_SIG.unique():\n",
    "    c_a = c.split(',')\n",
    "    clin_sigs.update(c_a)\n",
    "\n",
    "clin_sigs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will look at pathogenic tag"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Feature</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>...</th>\n",
       "      <th>AF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>rs1135172</td>\n",
       "      <td>1:11301714-11301714</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>MTOR</td>\n",
       "      <td>ENSG00000198793</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000361445.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>30487748,28977864</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>rs1135172</td>\n",
       "      <td>1:11301714-11301714</td>\n",
       "      <td>G</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001492400</td>\n",
       "      <td>promoter_flanking_region</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>30487748,28977864</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>rs1135172</td>\n",
       "      <td>1:11301714-11301714</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>MTOR</td>\n",
       "      <td>ENSG00000198793</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000361445.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>30487748,28977864</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>rs1135172</td>\n",
       "      <td>1:11301714-11301714</td>\n",
       "      <td>G</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00001492400</td>\n",
       "      <td>promoter_flanking_region</td>\n",
       "      <td>...</td>\n",
       "      <td>0.6396</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>30487748,28977864</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>209</th>\n",
       "      <td>rs6687605</td>\n",
       "      <td>1:25889632-25889632</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>LDLRAP1</td>\n",
       "      <td>ENSG00000157978</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000374338.4</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18772</th>\n",
       "      <td>rs1183768</td>\n",
       "      <td>9:135203231-135203231</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SETX</td>\n",
       "      <td>ENSG00000107290</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000224140.5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5561</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>25741868,24033266,23757202,25382069</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18773</th>\n",
       "      <td>rs1183768</td>\n",
       "      <td>9:135203231-135203231</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SETX</td>\n",
       "      <td>ENSG00000107290</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000372169.2</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5561</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>25741868,24033266,23757202,25382069</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18774</th>\n",
       "      <td>rs1183768</td>\n",
       "      <td>9:135203231-135203231</td>\n",
       "      <td>T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>SETX</td>\n",
       "      <td>ENSG00000107290</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000393220.1</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5561</td>\n",
       "      <td>benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>25741868,24033266,23757202,25382069</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18872</th>\n",
       "      <td>rs2229971</td>\n",
       "      <td>9:139407932-139407932</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000277541.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5278</td>\n",
       "      <td>likely_benign,benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>16729972</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18873</th>\n",
       "      <td>rs2229971</td>\n",
       "      <td>9:139407932-139407932</td>\n",
       "      <td>G</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>NOTCH1</td>\n",
       "      <td>ENSG00000148400</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000277541.6</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>...</td>\n",
       "      <td>0.5278</td>\n",
       "      <td>likely_benign,benign</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>16729972</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1600 rows Ã— 41 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      #Uploaded_variation               Location Allele  \\\n",
       "70              rs1135172    1:11301714-11301714      G   \n",
       "71              rs1135172    1:11301714-11301714      G   \n",
       "72              rs1135172    1:11301714-11301714      G   \n",
       "73              rs1135172    1:11301714-11301714      G   \n",
       "209             rs6687605    1:25889632-25889632      T   \n",
       "...                   ...                    ...    ...   \n",
       "18772           rs1183768  9:135203231-135203231      T   \n",
       "18773           rs1183768  9:135203231-135203231      T   \n",
       "18774           rs1183768  9:135203231-135203231      T   \n",
       "18872           rs2229971  9:139407932-139407932      G   \n",
       "18873           rs2229971  9:139407932-139407932      G   \n",
       "\n",
       "                     Consequence    IMPACT   SYMBOL             Gene  \\\n",
       "70            synonymous_variant       LOW     MTOR  ENSG00000198793   \n",
       "71     regulatory_region_variant  MODIFIER        -                -   \n",
       "72            synonymous_variant       LOW     MTOR  ENSG00000198793   \n",
       "73     regulatory_region_variant  MODIFIER        -                -   \n",
       "209             missense_variant  MODERATE  LDLRAP1  ENSG00000157978   \n",
       "...                          ...       ...      ...              ...   \n",
       "18772           missense_variant  MODERATE     SETX  ENSG00000107290   \n",
       "18773           missense_variant  MODERATE     SETX  ENSG00000107290   \n",
       "18774           missense_variant  MODERATE     SETX  ENSG00000107290   \n",
       "18872         synonymous_variant       LOW   NOTCH1  ENSG00000148400   \n",
       "18873         synonymous_variant       LOW   NOTCH1  ENSG00000148400   \n",
       "\n",
       "            Feature_type            Feature                   BIOTYPE  ...  \\\n",
       "70            Transcript  ENST00000361445.4            protein_coding  ...   \n",
       "71     RegulatoryFeature    ENSR00001492400  promoter_flanking_region  ...   \n",
       "72            Transcript  ENST00000361445.4            protein_coding  ...   \n",
       "73     RegulatoryFeature    ENSR00001492400  promoter_flanking_region  ...   \n",
       "209           Transcript  ENST00000374338.4            protein_coding  ...   \n",
       "...                  ...                ...                       ...  ...   \n",
       "18772         Transcript  ENST00000224140.5            protein_coding  ...   \n",
       "18773         Transcript  ENST00000372169.2            protein_coding  ...   \n",
       "18774         Transcript  ENST00000393220.1            protein_coding  ...   \n",
       "18872         Transcript  ENST00000277541.6            protein_coding  ...   \n",
       "18873         Transcript  ENST00000277541.6            protein_coding  ...   \n",
       "\n",
       "           AF              CLIN_SIG SOMATIC PHENO  \\\n",
       "70     0.6396                benign     0,1   1,1   \n",
       "71     0.6396                benign     0,1   1,1   \n",
       "72     0.6396                benign     0,1   1,1   \n",
       "73     0.6396                benign     0,1   1,1   \n",
       "209         -                     -       1     1   \n",
       "...       ...                   ...     ...   ...   \n",
       "18772  0.5561                benign     0,1   1,1   \n",
       "18773  0.5561                benign     0,1   1,1   \n",
       "18774  0.5561                benign     0,1   1,1   \n",
       "18872  0.5278  likely_benign,benign     0,1   1,1   \n",
       "18873  0.5278  likely_benign,benign     0,1   1,1   \n",
       "\n",
       "                                    PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS  \\\n",
       "70                       30487748,28977864          -         -            -   \n",
       "71                       30487748,28977864          -         -            -   \n",
       "72                       30487748,28977864          -         -            -   \n",
       "73                       30487748,28977864          -         -            -   \n",
       "209                                      -          -         -            -   \n",
       "...                                    ...        ...       ...          ...   \n",
       "18772  25741868,24033266,23757202,25382069          -         -            -   \n",
       "18773  25741868,24033266,23757202,25382069          -         -            -   \n",
       "18774  25741868,24033266,23757202,25382069          -         -            -   \n",
       "18872                             16729972          -         -            -   \n",
       "18873                             16729972          -         -            -   \n",
       "\n",
       "      MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  \n",
       "70                     -                     -  \n",
       "71                     -                     -  \n",
       "72                     -                     -  \n",
       "73                     -                     -  \n",
       "209                    -                     -  \n",
       "...                  ...                   ...  \n",
       "18772                  -                     -  \n",
       "18773                  -                     -  \n",
       "18774                  -                     -  \n",
       "18872                  -                     -  \n",
       "18873                  -                     -  \n",
       "\n",
       "[1600 rows x 41 columns]"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.SOMATIC != '-']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "#Uploaded_variation\n",
       "i5005436                                             pathogenic\n",
       "i5006568      likely_benign,benign/likely_benign,pathogenic,...\n",
       "i6015290      uncertain_significance,likely_benign,conflicti...\n",
       "i6015729                 pathogenic,benign/likely_benign,benign\n",
       "i6058764                                  protective,pathogenic\n",
       "i6060296      conflicting_interpretations_of_pathogenicity,b...\n",
       "rs10151259         not_provided,pathogenic,likely_benign,benign\n",
       "rs1024611                                pathogenic,risk_factor\n",
       "rs1042503                              likely_pathogenic,benign\n",
       "rs11085825                             benign,likely_pathogenic\n",
       "rs11558492                      pathogenic,likely_benign,benign\n",
       "rs16879498                                    benign,pathogenic\n",
       "rs1800435                              likely_benign,pathogenic\n",
       "rs2004640                                risk_factor,pathogenic\n",
       "rs2301612                                     pathogenic,benign\n",
       "rs2306283     conflicting_interpretations_of_pathogenicity,b...\n",
       "rs3755319                                            pathogenic\n",
       "rs460897                                pathogenic,_risk_factor\n",
       "rs4673                                 likely_pathogenic,benign\n",
       "rs6339                   benign,benign/likely_benign,pathogenic\n",
       "Name: CLIN_SIG, dtype: object"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_for = ['pathogenic']\n",
    "# df[df['CLIN_SIG'].str.contains('|'.join(search_for))]['#Uploaded_variation'].unique()\n",
    "pathogenic_df = df[df['CLIN_SIG'].str.contains('|'.join(search_for))].groupby('#Uploaded_variation').first()\n",
    "pathogenic_df = pathogenic_df[['Location', 'Allele', 'Codons', 'CLIN_SIG']]\n",
    "pathogenic_df.CLIN_SIG"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will find this SNPs in dbSNP https://www.ncbi.nlm.nih.gov/snp/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Codons</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>i5005436</th>\n",
       "      <td>6:32008312-32008312</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i5006568</th>\n",
       "      <td>2:71829924-71829924</td>\n",
       "      <td>G</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>likely_benign,benign/likely_benign,pathogenic,...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6015290</th>\n",
       "      <td>14:23887607-23887607</td>\n",
       "      <td>T</td>\n",
       "      <td>aaC/aaA</td>\n",
       "      <td>uncertain_significance,likely_benign,conflicti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6015729</th>\n",
       "      <td>1:156848918-156848918</td>\n",
       "      <td>T</td>\n",
       "      <td>Cat/Tat</td>\n",
       "      <td>pathogenic,benign/likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6058764</th>\n",
       "      <td>16:27356203-27356203</td>\n",
       "      <td>G</td>\n",
       "      <td>Atc/Gtc</td>\n",
       "      <td>protective,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6060296</th>\n",
       "      <td>19:13010520-13010520</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity,b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs10151259</th>\n",
       "      <td>14:21790040-21790040</td>\n",
       "      <td>T</td>\n",
       "      <td>Gct/Tct</td>\n",
       "      <td>not_provided,pathogenic,likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1024611</th>\n",
       "      <td>17:32579788-32579788</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1042503</th>\n",
       "      <td>12:103246700-103246700</td>\n",
       "      <td>T</td>\n",
       "      <td>gtG/gtA</td>\n",
       "      <td>likely_pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs11085825</th>\n",
       "      <td>19:13007458-13007458</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>benign,likely_pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs11558492</th>\n",
       "      <td>1:231408091-231408091</td>\n",
       "      <td>G</td>\n",
       "      <td>gAt/gGt</td>\n",
       "      <td>pathogenic,likely_benign,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs16879498</th>\n",
       "      <td>6:49580247-49580247</td>\n",
       "      <td>T</td>\n",
       "      <td>Gtt/Att</td>\n",
       "      <td>benign,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1800435</th>\n",
       "      <td>9:116153891-116153891</td>\n",
       "      <td>G</td>\n",
       "      <td>aaG/aaC</td>\n",
       "      <td>likely_benign,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2004640</th>\n",
       "      <td>7:128578301-128578301</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2301612</th>\n",
       "      <td>9:136301982-136301982</td>\n",
       "      <td>G</td>\n",
       "      <td>Caa/Gaa</td>\n",
       "      <td>pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2306283</th>\n",
       "      <td>12:21329738-21329738</td>\n",
       "      <td>G</td>\n",
       "      <td>Aat/Gat</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity,b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs3755319</th>\n",
       "      <td>2:234667582-234667582</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs460897</th>\n",
       "      <td>1:196716319-196716319</td>\n",
       "      <td>T</td>\n",
       "      <td>tCg/tTg</td>\n",
       "      <td>pathogenic,_risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4673</th>\n",
       "      <td>16:88713236-88713236</td>\n",
       "      <td>G</td>\n",
       "      <td>Tac/Cac</td>\n",
       "      <td>likely_pathogenic,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6339</th>\n",
       "      <td>1:156848946-156848946</td>\n",
       "      <td>T</td>\n",
       "      <td>gGt/gTt</td>\n",
       "      <td>benign,benign/likely_benign,pathogenic</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Location Allele   Codons  \\\n",
       "#Uploaded_variation                                           \n",
       "i5005436                6:32008312-32008312      T        -   \n",
       "i5006568                2:71829924-71829924      G  Atc/Gtc   \n",
       "i6015290               14:23887607-23887607      T  aaC/aaA   \n",
       "i6015729              1:156848918-156848918      T  Cat/Tat   \n",
       "i6058764               16:27356203-27356203      G  Atc/Gtc   \n",
       "i6060296               19:13010520-13010520      G        -   \n",
       "rs10151259             14:21790040-21790040      T  Gct/Tct   \n",
       "rs1024611              17:32579788-32579788      G        -   \n",
       "rs1042503            12:103246700-103246700      T  gtG/gtA   \n",
       "rs11085825             19:13007458-13007458      T        -   \n",
       "rs11558492            1:231408091-231408091      G  gAt/gGt   \n",
       "rs16879498              6:49580247-49580247      T  Gtt/Att   \n",
       "rs1800435             9:116153891-116153891      G  aaG/aaC   \n",
       "rs2004640             7:128578301-128578301      T        -   \n",
       "rs2301612             9:136301982-136301982      G  Caa/Gaa   \n",
       "rs2306283              12:21329738-21329738      G  Aat/Gat   \n",
       "rs3755319             2:234667582-234667582      C        -   \n",
       "rs460897              1:196716319-196716319      T  tCg/tTg   \n",
       "rs4673                 16:88713236-88713236      G  Tac/Cac   \n",
       "rs6339                1:156848946-156848946      T  gGt/gTt   \n",
       "\n",
       "                                                              CLIN_SIG  \n",
       "#Uploaded_variation                                                     \n",
       "i5005436                                                    pathogenic  \n",
       "i5006568             likely_benign,benign/likely_benign,pathogenic,...  \n",
       "i6015290             uncertain_significance,likely_benign,conflicti...  \n",
       "i6015729                        pathogenic,benign/likely_benign,benign  \n",
       "i6058764                                         protective,pathogenic  \n",
       "i6060296             conflicting_interpretations_of_pathogenicity,b...  \n",
       "rs10151259                not_provided,pathogenic,likely_benign,benign  \n",
       "rs1024611                                       pathogenic,risk_factor  \n",
       "rs1042503                                     likely_pathogenic,benign  \n",
       "rs11085825                                    benign,likely_pathogenic  \n",
       "rs11558492                             pathogenic,likely_benign,benign  \n",
       "rs16879498                                           benign,pathogenic  \n",
       "rs1800435                                     likely_benign,pathogenic  \n",
       "rs2004640                                       risk_factor,pathogenic  \n",
       "rs2301612                                            pathogenic,benign  \n",
       "rs2306283            conflicting_interpretations_of_pathogenicity,b...  \n",
       "rs3755319                                                   pathogenic  \n",
       "rs460897                                       pathogenic,_risk_factor  \n",
       "rs4673                                        likely_pathogenic,benign  \n",
       "rs6339                          benign,benign/likely_benign,pathogenic  "
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "#Uploaded_variation\n",
       "i3000469                                            risk_factor\n",
       "i6007787                                     risk_factor,benign\n",
       "i6058143                       drug_response,benign,risk_factor\n",
       "i6059141                       likely_benign,benign,risk_factor\n",
       "rs1024611                                pathogenic,risk_factor\n",
       "rs1049296                        risk_factor,association,benign\n",
       "rs1169288                                    benign,risk_factor\n",
       "rs12150220                                          risk_factor\n",
       "rs13266634                                          risk_factor\n",
       "rs1801197                                           risk_factor\n",
       "rs1801274                      drug_response,benign,risk_factor\n",
       "rs1801275                                           risk_factor\n",
       "rs1801394     uncertain_significance,benign,drug_response,ri...\n",
       "rs1801968                      likely_benign,benign,risk_factor\n",
       "rs2004640                                risk_factor,pathogenic\n",
       "rs2073658                                           risk_factor\n",
       "rs2184026                                           risk_factor\n",
       "rs2239704                                           risk_factor\n",
       "rs2241880                                    risk_factor,benign\n",
       "rs2281845                                    risk_factor,benign\n",
       "rs231775                                     benign,risk_factor\n",
       "rs4402960                                           risk_factor\n",
       "rs460897                                pathogenic,_risk_factor\n",
       "rs4880                         risk_factor,benign,drug_response\n",
       "rs4961                                risk_factor,drug_response\n",
       "rs5174                                              risk_factor\n",
       "rs5186                                       risk_factor,benign\n",
       "rs61747071                     risk_factor,benign,likely_benign\n",
       "rs6265                                       risk_factor,benign\n",
       "rs6280                                risk_factor,likely_benign\n",
       "rs6504649                                           risk_factor\n",
       "rs699                                        benign,risk_factor\n",
       "rs763110                                            risk_factor\n",
       "rs7794745                                           risk_factor\n",
       "rs909253                                            risk_factor\n",
       "Name: CLIN_SIG, dtype: object"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "search_for = ['risk_factor']\n",
    "# df[df['CLIN_SIG'].str.contains('|'.join(search_for))]['#Uploaded_variation'].unique()\n",
    "pathogenic_df = df[df['CLIN_SIG'].str.contains('|'.join(search_for))].groupby('#Uploaded_variation').first()\n",
    "pathogenic_df = pathogenic_df[['Location', 'Allele', 'Codons', 'CLIN_SIG']]\n",
    "pathogenic_df.CLIN_SIG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Codons</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>i3000469</th>\n",
       "      <td>2:138759649-138759649</td>\n",
       "      <td>T</td>\n",
       "      <td>aCa/aTa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6007787</th>\n",
       "      <td>2:234183368-234183368</td>\n",
       "      <td>G</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6058143</th>\n",
       "      <td>1:161479745-161479745</td>\n",
       "      <td>G</td>\n",
       "      <td>cAt/cGt</td>\n",
       "      <td>drug_response,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>i6059141</th>\n",
       "      <td>8:133909974-133909974</td>\n",
       "      <td>G</td>\n",
       "      <td>Atg/Gtg</td>\n",
       "      <td>likely_benign,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1024611</th>\n",
       "      <td>17:32579788-32579788</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>pathogenic,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1049296</th>\n",
       "      <td>3:133494354-133494354</td>\n",
       "      <td>T</td>\n",
       "      <td>Cct/Tct</td>\n",
       "      <td>risk_factor,association,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1169288</th>\n",
       "      <td>12:121416650-121416650</td>\n",
       "      <td>C</td>\n",
       "      <td>Atc/Ctc</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs12150220</th>\n",
       "      <td>17:5485367-5485367</td>\n",
       "      <td>T</td>\n",
       "      <td>cTc/cAc</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs13266634</th>\n",
       "      <td>8:118184783-118184783</td>\n",
       "      <td>T</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801197</th>\n",
       "      <td>7:93055753-93055753</td>\n",
       "      <td>G</td>\n",
       "      <td>cTg/cCg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801274</th>\n",
       "      <td>1:161479745-161479745</td>\n",
       "      <td>G</td>\n",
       "      <td>cAt/cGt</td>\n",
       "      <td>drug_response,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801275</th>\n",
       "      <td>16:27374400-27374400</td>\n",
       "      <td>G</td>\n",
       "      <td>cAg/cGg</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801394</th>\n",
       "      <td>5:7870973-7870973</td>\n",
       "      <td>G</td>\n",
       "      <td>atA/atG</td>\n",
       "      <td>uncertain_significance,benign,drug_response,ri...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs1801968</th>\n",
       "      <td>9:132580901-132580901</td>\n",
       "      <td>G</td>\n",
       "      <td>Gat/Cat</td>\n",
       "      <td>likely_benign,benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2004640</th>\n",
       "      <td>7:128578301-128578301</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,pathogenic</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2073658</th>\n",
       "      <td>1:161010762-161010762</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2184026</th>\n",
       "      <td>9:101304348-101304348</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2239704</th>\n",
       "      <td>6:31540141-31540141</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2241880</th>\n",
       "      <td>2:234183368-234183368</td>\n",
       "      <td>G</td>\n",
       "      <td>Act/Gct</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs2281845</th>\n",
       "      <td>1:201081943-201081943</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs231775</th>\n",
       "      <td>2:204732714-204732714</td>\n",
       "      <td>G</td>\n",
       "      <td>Acc/Gcc</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4402960</th>\n",
       "      <td>3:185511687-185511687</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs460897</th>\n",
       "      <td>1:196716319-196716319</td>\n",
       "      <td>T</td>\n",
       "      <td>tCg/tTg</td>\n",
       "      <td>pathogenic,_risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4880</th>\n",
       "      <td>6:160113872-160113872</td>\n",
       "      <td>G</td>\n",
       "      <td>gTt/gCt</td>\n",
       "      <td>risk_factor,benign,drug_response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs4961</th>\n",
       "      <td>4:2906707-2906707</td>\n",
       "      <td>T</td>\n",
       "      <td>Ggg/Tgg</td>\n",
       "      <td>risk_factor,drug_response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs5174</th>\n",
       "      <td>1:53712727-53712727</td>\n",
       "      <td>T</td>\n",
       "      <td>cGa/cAa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs5186</th>\n",
       "      <td>3:148459988-148459988</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs61747071</th>\n",
       "      <td>16:53720436-53720436</td>\n",
       "      <td>T</td>\n",
       "      <td>Gct/Act</td>\n",
       "      <td>risk_factor,benign,likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6265</th>\n",
       "      <td>11:27679916-27679916</td>\n",
       "      <td>T</td>\n",
       "      <td>Gtg/Atg</td>\n",
       "      <td>risk_factor,benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6280</th>\n",
       "      <td>3:113890815-113890815</td>\n",
       "      <td>T</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>risk_factor,likely_benign</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs6504649</th>\n",
       "      <td>17:48437456-48437456</td>\n",
       "      <td>G</td>\n",
       "      <td>aCa/aGa</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs699</th>\n",
       "      <td>1:230845794-230845794</td>\n",
       "      <td>G</td>\n",
       "      <td>aTg/aCg</td>\n",
       "      <td>benign,risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs763110</th>\n",
       "      <td>1:172627498-172627498</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs7794745</th>\n",
       "      <td>7:146489606-146489606</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>rs909253</th>\n",
       "      <td>6:31540313-31540313</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>risk_factor</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   Location Allele   Codons  \\\n",
       "#Uploaded_variation                                           \n",
       "i3000469              2:138759649-138759649      T  aCa/aTa   \n",
       "i6007787              2:234183368-234183368      G  Act/Gct   \n",
       "i6058143              1:161479745-161479745      G  cAt/cGt   \n",
       "i6059141              8:133909974-133909974      G  Atg/Gtg   \n",
       "rs1024611              17:32579788-32579788      G        -   \n",
       "rs1049296             3:133494354-133494354      T  Cct/Tct   \n",
       "rs1169288            12:121416650-121416650      C  Atc/Ctc   \n",
       "rs12150220               17:5485367-5485367      T  cTc/cAc   \n",
       "rs13266634            8:118184783-118184783      T  Cgg/Tgg   \n",
       "rs1801197               7:93055753-93055753      G  cTg/cCg   \n",
       "rs1801274             1:161479745-161479745      G  cAt/cGt   \n",
       "rs1801275              16:27374400-27374400      G  cAg/cGg   \n",
       "rs1801394                 5:7870973-7870973      G  atA/atG   \n",
       "rs1801968             9:132580901-132580901      G  Gat/Cat   \n",
       "rs2004640             7:128578301-128578301      T        -   \n",
       "rs2073658             1:161010762-161010762      T        -   \n",
       "rs2184026             9:101304348-101304348      T        -   \n",
       "rs2239704               6:31540141-31540141      C        -   \n",
       "rs2241880             2:234183368-234183368      G  Act/Gct   \n",
       "rs2281845             1:201081943-201081943      T        -   \n",
       "rs231775              2:204732714-204732714      G  Acc/Gcc   \n",
       "rs4402960             3:185511687-185511687      T        -   \n",
       "rs460897              1:196716319-196716319      T  tCg/tTg   \n",
       "rs4880                6:160113872-160113872      G  gTt/gCt   \n",
       "rs4961                    4:2906707-2906707      T  Ggg/Tgg   \n",
       "rs5174                  1:53712727-53712727      T  cGa/cAa   \n",
       "rs5186                3:148459988-148459988      C        -   \n",
       "rs61747071             16:53720436-53720436      T  Gct/Act   \n",
       "rs6265                 11:27679916-27679916      T  Gtg/Atg   \n",
       "rs6280                3:113890815-113890815      T  Ggc/Agc   \n",
       "rs6504649              17:48437456-48437456      G  aCa/aGa   \n",
       "rs699                 1:230845794-230845794      G  aTg/aCg   \n",
       "rs763110              1:172627498-172627498      T        -   \n",
       "rs7794745             7:146489606-146489606      T        -   \n",
       "rs909253                6:31540313-31540313      G        -   \n",
       "\n",
       "                                                              CLIN_SIG  \n",
       "#Uploaded_variation                                                     \n",
       "i3000469                                                   risk_factor  \n",
       "i6007787                                            risk_factor,benign  \n",
       "i6058143                              drug_response,benign,risk_factor  \n",
       "i6059141                              likely_benign,benign,risk_factor  \n",
       "rs1024611                                       pathogenic,risk_factor  \n",
       "rs1049296                               risk_factor,association,benign  \n",
       "rs1169288                                           benign,risk_factor  \n",
       "rs12150220                                                 risk_factor  \n",
       "rs13266634                                                 risk_factor  \n",
       "rs1801197                                                  risk_factor  \n",
       "rs1801274                             drug_response,benign,risk_factor  \n",
       "rs1801275                                                  risk_factor  \n",
       "rs1801394            uncertain_significance,benign,drug_response,ri...  \n",
       "rs1801968                             likely_benign,benign,risk_factor  \n",
       "rs2004640                                       risk_factor,pathogenic  \n",
       "rs2073658                                                  risk_factor  \n",
       "rs2184026                                                  risk_factor  \n",
       "rs2239704                                                  risk_factor  \n",
       "rs2241880                                           risk_factor,benign  \n",
       "rs2281845                                           risk_factor,benign  \n",
       "rs231775                                            benign,risk_factor  \n",
       "rs4402960                                                  risk_factor  \n",
       "rs460897                                       pathogenic,_risk_factor  \n",
       "rs4880                                risk_factor,benign,drug_response  \n",
       "rs4961                                       risk_factor,drug_response  \n",
       "rs5174                                                     risk_factor  \n",
       "rs5186                                              risk_factor,benign  \n",
       "rs61747071                            risk_factor,benign,likely_benign  \n",
       "rs6265                                              risk_factor,benign  \n",
       "rs6280                                       risk_factor,likely_benign  \n",
       "rs6504649                                                  risk_factor  \n",
       "rs699                                               benign,risk_factor  \n",
       "rs763110                                                   risk_factor  \n",
       "rs7794745                                                  risk_factor  \n",
       "rs909253                                                   risk_factor  "
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pathogenic_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
